### PR TITLE
Fix ImportFile() to send a multipart form request

### PR DESCRIPTION
**Pull request by dheitman42**
Update ImportFile() to send a multipart form request, add missing "Name" parameter to ImportFileOptions.

### CODE REVIEW

**Comment by dheitman42** [comment]
I'll test this more thoroughly after the weekend.

**Comment by dheitman42** [comment]
Note this is intended to resolve https://github.com/xanzy/go-gitlab/issues/636

File: project_import_export.go [hunk 1]
```diff
@@ -173,10 +179,41 @@ type ImportFileOptions struct {
...
+		if strings.ToLower(key) == "file" {
+			part, err := writer.CreateFormFile("file", file.Name())
+			if err != nil {
+				return nil, nil, err
+			}
+			_, err = io.Copy(part, file)
+		} else {
+			_ = writer.WriteField(strings.ToLower(key), fmt.Sprintf("%v", value))
+		}
+	}
```
Comment (svanharmelen) [inline project_import_export.go:206]: No clue what you are trying to do here, but this doesn't look right to me. Can you please elaborate on this approach?

Comment (dheitman42) [reply project_import_export.go:206]: Sure, we're creating a multipart form, creating the file component with CreateFormFile(), copying the file contents to that part, then adding each of the fields from the ImportFileOptions struct as additional field data in the multipart form.  That's the intent, anyway.  I'll admit to being n00bish with golang, so maybe this isn't the proper way to do the struct-to-string-map conversion, but that's the basic idea.

Comment (dheitman42) [reply project_import_export.go:206]: Dunno if that answered your question, happy to clarify further if need be.

**Comment by svanharmelen** [comment]
This should be fixed by PR #1318 which also adds some additional file upload improvements. Please open an issue if you have any issues with the new code. Thanks for the PR and the discussions!

**Pull request closed without merging**